server/command/util.go

```diff
@@ -0,0 +1,30 @@
+package command
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func parseServerURL(u string, withPort bool) (string, error) {
+	ur, err := url.Parse(u)
+	if err != nil {
+		return "", errors.Wrap(err, "bad url")
+	}
+	if ur.Scheme != "http" && ur.Scheme != "https" {
+		return "", errors.New("bad scheme")
+	}
+	ur.Scheme = "https"
+
+	host := ur.Host
+	if strings.Contains(host, ":") {
+		i := strings.LastIndex(host, ":")
+		host = host[:i]
+	}
+	if withPort {
+		host = host + ":8089"
+	}
```

Comment:
This block runs when `noport` is not provided, so it implicitly adds `:8089` in the default case. What if the admin wants to provide a specific port?